02-1.py

```diff
@@ -0,0 +1,17 @@
+from utils import read_inputs
+
+inputs = read_inputs('inputs/input2.txt')
+
+right_pswd = 0
+
+for ele in inputs:
+	temp_tab = ele.split(' ')
+	tab_num = temp_tab[0].split('-')
```

Comment:
Another solution for these kind of problem (parsing a string) would be to create a regular expression (regex). They are often slower but for more complex cases they can be handy ;) 
I prefer your way of doing think, I just wanted to mention it just in case you didn't know ;)

Reply:
If you cut down an entity into several smaller fragment (like your doing here with a string), it is often clearer to just give name to the entities. 
`rules, letter_and_password = ele.split(' ')`
`min_allowed, max_allowed = rules.split('-')`
`letter_of_interest, password = letter_and_password.split(':')`

Reply:
When you do the first line, python considers that rules will be the first element and letter_and_password will be fill with the rest right? Can we like invert the process and say the rules should take all elements but the ast and letter_and_password only the last one with the same kind of logic?

Reply:
My bad, I though there was only one " " caracter. 
`rules, letter, password = ele.split(' ')`
`letter = letter[0]` # remove the `:`

Reply:
I think python will just crash if you do `rules, letter_and_password = ele.split(' ')` because it expect 3 elements